velox/expression/EvalCtx.h

```diff
@@ -27,8 +27,139 @@ namespace facebook::velox::exec {
 
 class Expr;
 class ExprSet;
+class LocalDecodedVector;
+class LocalSelectivityVector;
 struct ScopedContextSaver;
 
+/// Utility class for enabling peeling functionality. It takes a list of
```

Comment:
the comments clarified many corner cases I hadn't really digested before about how peelling works. Thanks for cleaning this up!